src/widgets/macro_view.rs

```diff
@@ -107,44 +223,41 @@ impl MacroTextBox {
                 let notification = if config.notification_config.different_last_notification
                     && index == max_index
                 {
-                    &config.notification_config.custom_last_notification_format
+                    &fixed_formatting_data.last_notification_format
                 } else {
-                    &config.notification_config.custom_notification_format
+                    &fixed_formatting_data.notification_format
                 };
 
-                lines.push(format_custom_notification(notification, index, max_index))
+                lines.push(format_custom_macro_command(notification, index, max_index))
             }
         }
         Self {
             text: lines.join(newline),
+            id: egui::Id::new(("MACRO_TEXT", index)),
```

Comment:
This is a much nicer way of generating an ID. I didn't realize this was possible :)